### ISSUE TITLE
Is it expected behaviour that Shlink will delete any and all keys in the Redis server?

### ISSUE THREAD

**Issue by frankfil** [shlinkio/shlink#1580]
#### How Shlink is set-up

* Shlink Version: 3.3.2 
* How do you serve Shlink: Docker image
* Database engine used: MySQL

#### Summary

Is it the expected behaviour, or is it a bug, that Shlink when configured to use Redis will delete ALL the Keys in Redis even ones it did not create?

In other words any existing Keys in Redis get deleted by Shlink almost immediately after starting up.

I could not find anything in the docs indicating that this would occur.

If this is the intended behaviour it might be nice to point this out in the docs lest someone find out the hard way!

**Comment by acelaya** [comment]
No, that's not the expected behavior. There's no reason for Shlink to do that.

I will check it.

**Comment by milkogpetrov** [comment]
I think this might be a bug. I attached the log from redis-cli monitor after starting/restarting shlink instance. 

Check line 95: ("1670580633.643543 [0 172.27.0.6:40652] "FLUSHDB")

[redis-monitor-shlink-start.log](https://github.com/shlinkio/shlink/files/10193823/redis-monitor-shlink-start.log)

My set-up (all applications run as docker containers):
- shlink-1 (ver 3.3.2)
- shlink-2 (ver 3.3.2)
- mariadb (ver 10.8)
- redis (ver 6.2)

**Comment by acelaya** [comment]
Wow, I would not expect that to happen. I wonder if [this](https://github.com/shlinkio/shlink/blob/2047d6b77233b58bf4b94c106a9d8b423aa649ff/docker/docker-entrypoint.sh#L19) is the culprit. Need to check further what does that command actually do

**Comment by acelaya** [comment]
Ok, yes, the command above does run a FLUSHDB when configured cache is redis and there's no cache namespace configured.

Apparently the cache namespace was removed from Shlink in some version. I will get it back to fix this.

**Comment by acelaya** [comment]
I have just merged the fix for this. It will be included with Shlink 3.4.0
